pkg/common/cns-lib/volume/manager.go

```diff
@@ -799,9 +804,20 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 	return resp, faultType, err
 }
 
+func ensureOperationContextHasATimeout(ctx context.Context) (context.Context, context.CancelFunc) {
+	_, ok := ctx.Deadline()
+	if !ok {
+		// no timeout is set, so we need to set it
+		return context.WithTimeout(ctx, DefaultOperationTimeoutInSeconds*time.Second)
```

Comment:
if we do so, then logger created out of context will get disassociated. Are you sure tracing is working fine with this change?

Reply:
```
log2 := logger.GetLogger(ctx)
log2.Infof("before checking timeout")
ctx, cancelFunc := ensureOperationContextHasATimeout(ctx)
defer cancelFunc()
log3 := logger.GetLogger(ctx)
log3.Infof("after checking for timeout")

internalCreateVolume := func() (*CnsVolumeInfo, string, error) {
	log := logger.GetLogger(ctx)
	...
}

start := time.Now()
resp, faultType, err := internalCreateVolume()
log := logger.GetLogger(ctx)
log.Debugf("internalCreateVolume: returns fault %q", faultType)

---

2023-08-04T00:52:42.981Z	INFO	wcp/controller.go:836	CreateVolume: called with args {Name:pvc-53013d35-0b49-45bd-9840-56ca8f7ee101 CapacityRange:required_bytes:1073741824  VolumeCapabilities:[mount:<fs_type:"ext4" > access_mode:<mode:SINGLE_NODE_WRITER > ] Parameters:map[storagePolicyID:8241f639-23b9-4862-9036-f8ff9fdacf7e] Secrets:map[] VolumeContentSource:<nil> AccessibilityRequirements:requisite:<segments:<key:"kubernetes.io/hostname" value:"4216cdc06da1f15245b3f4d95feba699" > > requisite:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-49-177.eng.vmware.com" > > requisite:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-51-199.eng.vmware.com" > > requisite:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-52-67.eng.vmware.com" > > requisite:<segments:<key:"kubernetes.io/hostname" value:"421683f77c0d860f2bed7e67f443739a" > > requisite:<segments:<key:"kubernetes.io/hostname" value:"4216924008dc1abbd9bcf98b616e40f5" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-52-67.eng.vmware.com" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"421683f77c0d860f2bed7e67f443739a" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"4216924008dc1abbd9bcf98b616e40f5" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"4216cdc06da1f15245b3f4d95feba699" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-49-177.eng.vmware.com" > > preferred:<segments:<key:"kubernetes.io/hostname" value:"sc2-10-186-51-199.eng.vmware.com" > >  XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}

2023-08-04T00:52:43.269Z	INFO	volume/manager.go:770	before checking timeout	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}
2023-08-04T00:52:43.271Z	INFO	volume/manager.go:774	after checking for timeout	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}

2023-08-04T00:52:44.828Z	INFO	volume/util.go:326	Volume created successfully. VolumeName: "pvc-53013d35-0b49-45bd-9840-56ca8f7ee101", volumeID: "b12d68e1-15b6-4cf4-b080-c457614ae009"	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}

2023-08-04T00:52:44.829Z	DEBUG	volume/util.go:328	CreateVolume volumeId {{} "b12d68e1-15b6-4cf4-b080-c457614ae009"} is placed on datastore "ds:///vmfs/volumes/vsan:521f9f32c2471e97-dc72780e110ffd43/"	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}

2023-08-04T00:52:44.869Z	DEBUG	volume/manager.go:799	internalCreateVolume: returns fault ""	{"TraceId": "abf67e2b-5ae9-4d8e-b4a0-8bf3d0180b82"}
```

Reply:
@divyenpatel, tracing is preserved. I've called getlogger multiple times before and after the method I've added. All logs have the same traceID